http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java

```diff
@@ -15,31 +15,77 @@
 
 package software.amazon.awssdk.http.nio.netty;
 
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFactory;
 import io.netty.channel.EventLoopGroup;
-import io.netty.channel.epoll.EpollEventLoopGroup;
 import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioSocketChannel;
 import java.util.Optional;
 import java.util.concurrent.ThreadFactory;
 import software.amazon.awssdk.annotations.SdkPublicApi;
 import software.amazon.awssdk.utils.ThreadFactoryBuilder;
-import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
 import software.amazon.awssdk.utils.builder.CopyableBuilder;
 import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
 
 /**
- * Configurable factory for creating {@link EventLoopGroup} instances. Will choose the optimal implementation of
- * {@link EventLoopGroup} per the platform (i.e. if on Linux then {@link EpollEventLoopGroup} will be used).
+ * Provides {@link EventLoopGroup} and {@link ChannelFactory} for {@link NettyNioAsyncHttpClient}.
+ *
+ * <p>
+ * There are two ways to create a new instance.
+ *
+ * <ul>
+ * <li>Using {@link #create(EventLoopGroup, ChannelFactory)} to provide a custom {@link EventLoopGroup} and
+ * {@link ChannelFactory}</p>.
+ * The {@link EventLoopGroup} <b>MUST</b> be closed by the caller when it is ready to
+ * be disposed. The SDK will not close the {@link EventLoopGroup} when the HTTP client is closed. See
+ * {@link EventLoopGroup#shutdownGracefully()} to properly close the event loop group.
+ * </li>
+ *
+ * <li>using {@link #builder()} to provide custom configuration of {@link EventLoopGroup}. The {@link EventLoopGroup} created by
+ * the builder is managed by the SDK and will be shutdown when the HTTP client is closed.
+ * </li>
+ * </ul>
  */
 @SdkPublicApi
-public final class DefaultEventLoopGroupFactory
-        implements ToCopyableBuilder<DefaultEventLoopGroupFactory.Builder, DefaultEventLoopGroupFactory>, EventLoopGroupFactory {
+public final class SdkEventLoopGroup implements ToCopyableBuilder<SdkEventLoopGroup.Builder, SdkEventLoopGroup> {
 
+    private final EventLoopGroup eventLoopGroup;
+    private final ChannelFactory<? extends Channel> channelFactory;
     private final Integer numberOfThreads;
```

Comment:
Do we need to have fields for number of threads and thread factory or can we just use them to create an event loop group at build time?

Reply:
I was thinking customers can retrieve the value by calling `numberOfThreads()`. But that's not necessary and can be confusing to customers who use `SdkEventLoopGroup.create` method to create the class. Removed.